pkg/sql/pgwire/pgwirebase/too_big_error.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 The Cockroach Authors.
```

Comment:
I am impressed by how much effort went into the code in this file!
I will make comments about your implementation below, for this and the next time you want to implement a custom error.

However, meanwhile I believe you don't actually need this machinery for this PR. I don't even think you need a custom error type. A pgwire datadriven test will tell your test everything it needs to know.

Reply:
I do want one; I need to differentiate errors between:

```
readConn, err := io.ReadAll(...)
// error during streaming here
```

as opposed to the case where we encountered a big size.